src/main/java/fi/aalto/cs/apluscourses/presentation/base/BaseListViewModel.java

```diff
@@ -0,0 +1,114 @@
+package fi.aalto.cs.apluscourses.presentation.base;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.swing.AbstractListModel;
+import javax.swing.DefaultListSelectionModel;
+import javax.swing.ListModel;
+import javax.swing.ListSelectionModel;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An abstract extension to {@link ListModel} that co-operates with {@link ListElementViewModel}.
+ * Provides the following functionality:
+ * <ul>
+ * <li>selection state to be stored inside the element objects</li>
+ * <li>the view to be informed about changes in the presentation model</li>
+ * </ul>
+ * @param <E> Type of the list elements (presentation models), subtype of
+ *            {@link ListElementViewModel}
+ */
+public class BaseListViewModel<E extends ListElementViewModel<?>> extends AbstractListModel<E>
+    implements SelectableListModel<E> {
+
+  @NotNull
+  private final transient ListSelectionModel selectionModel;
+
+  // Not actually something that should be transient but we are not doing serialization and Sonar
+  // requires fields be marked transient if they are not serializable.
+  @NotNull
+  private final transient List<E> elements;
```

Comment:
does this apply to all the `transient`-marked cases within the CL?

Reply:
Yes. And now when I think about it, the model classes could be declared serializable (not now but at some point when the software and the whole world are otherwise ready).

So I remove the transient keywords and let Sonar keep reminding us of this (non-)issue which can then be "fixed" by declaring model classes serializable at some point.